### PR TITLE
Created voting table migration, model and updates API endpoints and test

### PR DESCRIPTION
**Pull request by acambece**
Dan:

Here's the changelog:
- Created migration for vote tables
- Updated associations for Reviews, Cities, Users and votes via models
- Updated tests to confirm returned json data

Thanks!

### CODE REVIEW

File: app/controllers/api/v1/reviews_controller.rb [hunk 1]
```diff
@@ -1,7 +1,33 @@
...
+          hash[:upvotes] = hash[:upvotes] + 1
+        elsif v[:vote] == false
+          hash[:downvotes] = hash[:downvotes] + 1
+        end
+      end
+      city_reviews << hash
+    end
     render json: city_reviews
   end
 end
```
Comment (nwalberts) [inline app/controllers/api/v1/reviews_controller.rb]: this is a bit janky, and needs refactoring. Constructing your own hash like this is very tedious, and using a JSON serializer is preferred. You should also consider a private method or two at the bottom of this file to help organize it and set the upvote and downvote correctly.

Comment (nwalberts) [reply app/controllers/api/v1/reviews_controller.rb]: You also should utilize your activerecord associations to grab the correct reviews for the city in question

File: app/controllers/api/v1/users_controller.rb [hunk 2]
```diff
@@ -5,12 +5,37 @@ class Api::V1::UsersController < ApplicationController
...
+        @reviews << hash
+      end
+
       @cities = City.where(user_id: @current_user.id)
       render json: {current_user: @current_user, reviews: @reviews, cities: @cities}
     else
       render json: {current_user: nil, message: "Please sign in.", status: 401}
     end
   end
-
```
Comment (nwalberts) [inline app/controllers/api/v1/users_controller.rb]: Same thing here, needs a good deal of refactoring here. Use a serializer

File: app/models/user.rb [hunk 3]
```diff
@@ -1,4 +1,7 @@
 class User < ApplicationRecord
+  has_many :cities
+  has_many :reviews, through: :cities
+  has_many :votes, through: :reviews
   mount_uploader :avatar, AvatarUploader
   # Include default devise modules. Others available are:
```
Comment (nwalberts) [inline app/models/user.rb]: The last two has_many associations should not need the `through` keywords. 
E.g. there is a direct relationship between a user and a review, so the through keyword is not needed

File: db/migrate/20180126150047_create_vote_system.rb [hunk 4]
```diff
@@ -0,0 +1,9 @@
+class CreateVoteSystem < ActiveRecord::Migration[5.1]
+  def change
+    create_table :votes do |t|
+      t.belongs_to :user, null: false
+      t.belongs_to :review, null: false
+      t.boolean :vote
```
Comment (nwalberts) [inline db/migrate/20180126150047_create_vote_system.rb]: While not required, I would strongly suggesting avoiding a boolean for an attribute that can have three values (upvoted, downvoted, or not voted). `nil` as a value can lead to errors and bugs.

File: app/models/review.rb [hunk 5]
```diff
@@ -6,4 +6,23 @@ class Review < ApplicationRecord
   validates :comfort_index, presence: true, numericality: true, format: { with: /(?<!\S)[1-5](?!\S)/, message: "must be a number between 1 and 5" }
   validates :weather_variance, presence: true, numericality: true, format: { with: /(?<!\S)[1-5](?!\S)/, message: "must be a number between 1 and 5" }
 
+  def upvotes
```
Comment (dpickett) [inline app/models/review.rb:9]: if you define this routine in the model, you do not need to duplicate it in the correlating serializer. You should be able to define these methods once.

**Review by dpickett — CHANGES_REQUESTED** [review]

**Review by dpickett — APPROVED** [review]

**Pull request merged**